.github/workflows/codecov.yaml

```diff
@@ -0,0 +1,106 @@
+name: Run tests and upload coverage
+
+on: [push, pull_request]
+
+jobs:
+  test:
+    name: Run tests and collect coverage
+    runs-on: ubuntu-latest
+    # https://github.com/bahmutov/cypress-gh-action-split-install/blob/ca3916d4e7240ebdc337825d2d78eb354855464b/.github/workflows/tests.yml#L8-L11
+    env:
+      # prevents extra Cypress installation progress messages
+      CI: 1
+      # avoid warnings like "tput: No value for $TERM and no -T specified"
+      TERM: xterm
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 2
+
+      - name: Install pnpm
+        uses: pnpm/action-setup@v4
+        with:
+          run_install: false
+
+      - name: Set up Node
+        uses: actions/setup-node@v4
+        with:
+          node-version: '22'
+          cache: pnpm
+
+      - name: Install dependencies
+        run: pnpm install
+
+      - name: Building packages
+        run: pnpm build
+
+      - name: Run Jest tests with coverage
+        run: npx jest --coverage --coverageDirectory=./coverage/jest
+        continue-on-error: true
+
+      - name: Upload Jest coverage to Codecov
+        uses: codecov/codecov-action@v5
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
+          file: ./coverage/jest/lcov.info
+          name: jest-coverage
+        if: always()
```

Comment:
Instead of a separate code-coverage-specific workflow, I think it would make more sense to upload the coverage as part of the test workflow that already exists to avoid duplication and not have to spin up another worker instance, specifically [here](https://github.com/commercetools/merchant-center-application-kit/blob/e115255feebe89379a9cf8b175540b93fa63a30f/.github/workflows/main.yml#L47).

Reply:
Done